src/borg/archive.py

```diff
@@ -519,7 +519,7 @@ def extract_helper(self, dest, item, path, stripped_components, original_path, h
                 # assign chunks to this item, since the item which had the chunks was not extracted
                 item.chunks = chunks
         yield hardlink_set
-        if not hardlink_set and hardlink_masters:  # 2nd term, is it correct/needed?
```

Comment:
btw, it can not be None here, due to https://github.com/borgbackup/borg/pull/2342/commits/155f38c2333b3892527796d96e999149d253616e#diff-a7aee10470e434998489f84fe93dc2bcR541